### PR TITLE
Add custom_client_ip_header and tls_policy to custom_domain resource

### PR DESCRIPTION
**Pull request by cv711**
### 🔧 Changes

This PR is exposing the `custom_client_ip_header` [option](https://auth0.com/docs/api/management/v2/#!/Custom_Domains/post_custom_domains) for custom domains

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

**Comment by cv711** [comment]
So I'm trying to record some new http interactions and I'm facing a couple of problems:

- The documentation doesn't describe that [this](https://github.com/auth0/terraform-provider-auth0/blob/main/internal/provider/resource_auth0_custom_domain_test.go#L51) check has to be commented out in order to record http interations for other domains.
- After doing that and trying to proceed with recording a new http interaction I get the following error:
```
--- FAIL: TestAccCustomDomain (5.77s)
    resource_auth0_custom_domain_test.go:56: Step 1/1 error: Error running apply: exit status 1
        
        Error: failed to send the request: Post "https://babylon-identity-test.eu.auth0.com/api/v2/custom-domains": requested interaction not found
        
          with auth0_custom_domain.my_custom_domain,
          on terraform_plugin_test.tf line 2, in resource "auth0_custom_domain" "my_custom_domain":
           2: resource "auth0_custom_domain" "my_custom_domain" {
        
FAIL
FAIL    github.com/auth0/terraform-provider-auth0/internal/provider     6.517s
FAIL
make: *** [test-acc] Error 1
```
- So I went ahead and manually added the missing field in the recording, since I don't have access to the domain the tests where created with.
- If someone could update the recording for me with the correct process we should be ready to go with this one. 
Thanks!

**Comment by sergiught** [comment]
Hey @cv711, I set this branch to draft mode while we get this in the right place.

File: internal/provider/resource_auth0_custom_domain.go [hunk 1]
```diff
@@ -74,14 +74,34 @@ func newCustomDomain() *schema.Resource {
 					},
 				},
 			},
+			"custom_client_ip_header": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"cf-connecting-ip", "x-forwarded-for", "true-client-ip", "",
```
Comment (sergiught) [inline internal/provider/resource_auth0_custom_domain.go:81]: We're adding as well validation as this field can only have certain values.

File: internal/provider/resource_auth0_custom_domain.go [hunk 2]
```diff
@@ -74,14 +74,34 @@ func newCustomDomain() *schema.Resource {
...
+				}, false),
+				Description: "The HTTP header to fetch the client's IP address. " +
+					"Cannot be set on auth0_managed domains.",
+			},
+			"tls_policy": {
```
Comment (sergiught) [inline internal/provider/resource_auth0_custom_domain.go:86]: We're adding this field as well for completeness as the PATCH endpoint accepts it as well.

File: internal/provider/resource_auth0_custom_domain_test.go [hunk 3]
```diff
@@ -47,35 +46,135 @@ func init() {
 	})
 }
 
-func TestAccCustomDomain(t *testing.T) {
-	if os.Getenv("AUTH0_DOMAIN") != recorder.RecordingsDomain {
-		t.Skip()
-	}
+const testAccCreateSelfManagedCustomDomain = `
```
Comment (sergiught) [inline internal/provider/resource_auth0_custom_domain_test.go:49]: Refactoring the tests so they're covering more cases.

File: internal/provider/resource_auth0_custom_domain_test.go [hunk 4]
```diff
@@ -47,35 +46,135 @@ func init() {
 	})
 }
 
-func TestAccCustomDomain(t *testing.T) {
-	if os.Getenv("AUTH0_DOMAIN") != recorder.RecordingsDomain {
```
Comment (sergiught) [inline internal/provider/resource_auth0_custom_domain_test.go:51]: It turns out we didn't really need to run these tests only on the predefined recordings, we can run them on the live tenant on demand as well without anything breaking as we're not going through the verification.

File: internal/provider/resource_auth0_custom_domain.go [hunk 5]
```diff
@@ -135,9 +172,18 @@ func deleteCustomDomain(ctx context.Context, d *schema.ResourceData, m interface
...
+func expandCustomDomain(d *schema.ResourceData) *management.CustomDomain {
+	config := d.GetRawConfig()
+
+	customDomain := &management.CustomDomain{
+		TLSPolicy:            value.String(config.GetAttr("tls_policy")),
+		CustomClientIPHeader: value.String(config.GetAttr("custom_client_ip_header")),
 	}
+
+	if d.IsNewResource() {
+		customDomain.Domain = value.String(config.GetAttr("domain"))
```
Comment (sergiught) [inline internal/provider/resource_auth0_custom_domain.go:184]: We still wanna make sure the domain and type can only be set on a new resource but are not getting sent on the PATCH.

**Review by sergiught — COMMENTED** [review]
Adding a few comments to ease up reviewing, also to give you an overview of all the extra changes that were still needed @cv711 . Hope it helps for future contributions from your part! ☘️ Looking forward to reviewing some more of your work:)

**Comment by sergiught** [comment]
@cv711 If you wanna test this as well on your side before we merge this and cut a release you can use the `make install VERSION=XX.XX.XX` command and then simply update your terraform config with the correct installed version. It will fetch it from local and you can use it. 

e.g. 

```
terraform {
  required_providers {
    auth0 = {
      source  = "auth0/auth0"
      version = "XX.XX.XX"
    }
  }
}
```

File: internal/provider/resource_auth0_custom_domain.go (hunk 2 above)
Comment (cv711) [reply internal/provider/resource_auth0_custom_domain.go:86]: I was planning to add this field in my next PR but good to have them both here!

**Comment by cv711** [comment]
🚀 all good from my side! Thank you very much for your help!

**Review by sergiught — APPROVED** [review]

**Pull request merged**